### DISCUSSION TITLE
How can I parse "winget list" output without trimmed IDs

### DISCUSSION THREAD

**Discussion by MichaelBrunn3r** [microsoft/winget-cli#4090] [General] [answered]
I'm trying to write a small CLI tool that parses the output from "winget list" and converts it into another format. Unfortunatelly I found out that winget formats its output depending on the console size, so calling it as a child process limits it to 120 columns. IDs are trimmed which makes this useless for this usecase.
Is threre a way to get the output without trimmed columns?

I saw that PSWinGet somehow manages to parse the output independent of the console size, even though its just parses the raw string output. I was suggested using the COM API instead, but I hit some roadblocks.

**Comment by mdanish-kh** [comment, marked as answer]
Have you tried using the `Get-WinGetPackage` cmdlet in the [Microsoft.WinGet.Client PowerShell Module](https://www.powershellgallery.com/packages/Microsoft.WinGet.Client/)?

For getting the IDs from `winget list`, the PowerShell command would be
```
Get-WinGetPackage | Select Id
```

**Reply by MichaelBrunn3r** [reply]
Oh, sorry my question wasn't clear enough. Yes, I have tried the Powershell Module, but I wanted to write a module for Nushell, which requires me to use Rust. PSWinget and WingetUI (written in Python) both just call `winget list` and parse that, but they somehow don't get formatting issues.
I thought I could just copy the code from PSWinget and execute it: `pwsh --command "Invoke-Command \"winget list\""`. But the issue remains that this only seems to work if I am executing the command from within a shell.

I don't think I should ask Rust related issues here, so I thought there might just be a language agnostic solution.

**Comment by Martin-Widmer** [comment]
Were you able to solve this? I ran into the same problem.

**Comment by denelon** [comment]
For anything automated, I'd suggest looking at the WinGet COM APIs, or use cmdlets from one of the WinGet PowerShell modules. The WinGet CLI was designed for interactive use by a logged in user.
